### PR TITLE
Don't check for ConfigurationOptions equality during ConfigurationNode equality check

### LINKED ISSUE #114: ConfigurationOptions is checked for equality when checking ConfigurationNode equality
It seems to me that there isn't a particular reason for this. If someone is checking for ConfigurationNode equality they're almost always intending to check when the values are equal and (to a lesser importance) if the keys are equal.

In my opinion, `loader.load().equals(loader.load())` should be able to return `true`, and yet it doesn't. Given how ConfigurationOptions are created, `equals` may as well literally be a `==` check.

### PR DESCRIPTION
**Pull request by FerusGrim**
I was expecting there to be more to this, but it doesn't appear as though there are any changes required elsewhere to fix this.

Closes #114

### CODE REVIEW

**Comment by lucko** [comment]
That whole method can probably be replaced by

```java
return Objects.hashCode(key) ^ value.hashCode();
```

**Comment by FerusGrim** [comment]
@lucko Implemented the requested change. Silly mistake.

File: configurate-core/src/main/java/ninja/leaping/configurate/SimpleConfigurationNode.java [hunk 1]
```diff
@@ -596,16 +596,12 @@ public boolean equals(Object o) {
         SimpleConfigurationNode that = (SimpleConfigurationNode) o;
 
         return Objects.equals(this.key, that.key) &&
-                this.options.equals(that.options) &&
                 this.value.equals(that.value);
```
Comment (kashike) [inline configurate-core/src/main/java/ninja/leaping/configurate/SimpleConfigurationNode.java]: use `Objects.equals(this.value, that.value)` for consistency

**Comment by FerusGrim** [comment]
@kashike Also fulfilled your request.

**Pull request merged**